src/main/java/io/strimzi/utils/LogicalKafkaVersionEntity.java

```diff
@@ -0,0 +1,192 @@
+            String[] otherComponents = version2.split("\\.");
+            for (int i = 0; i < Math.min(components.length, otherComponents.length); i++) {
+                int x = Integer.parseInt(components[i]);
+                int y = Integer.parseInt(otherComponents[i]);
+                // equality of version is ignored...
+                if (x < y) {
+                    return -1;
+                } else if (x > y) {
+                    return 1;
+                }
+
+            }
+            return components.length - otherComponents.length;
+        }
+
+        // f.e. "3.0.0"
+        public String getVersion() {
+            return version;
+        }
+
+        // f.e. "test-container:0.1.0-kafka-3.0.0
+        public String getImage() {
+            return image;
+        }
+
+        public String getStrimziTestContainerVersion() {
+            Matcher regex = STRIMZI_TEST_CONTAINER_IMAGE_WITHOUT_KAFKA_VERSION.matcher(this.image);
+
+            // only one occurrence in the string
+            if (regex.find()) {
+                final String strimziContainerVersion = regex.group(1);
+                LOGGER.info("Find the version of Strimzi kafka container:{}", strimziContainerVersion);
+                return strimziContainerVersion;
+            } else {
+                throw new RuntimeException("Unable to find version of Strimzi kafka container from image name - " + this.image);
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "LogicalKafkaVersionEntity{" +
+                "version='" + this.version + '\'' +
+                ", image='" + this.image + '\'' +
+                '}';
+        }
+    }
+
+    /**
+     * Get the latest release
+     * @return LogicalKafkaVersion latest release
+     */
+    public LogicalKafkaVersion latestRelease() {
+        // at least one release in the json schema is needed
+        if (this.logicalKafkaVersionEntities == null || this.logicalKafkaVersionEntities.size() < 1) {
+            throw new IllegalStateException("Wrong json schema! It must have at least one release");
+        }
+
+        LogicalKafkaVersion latestRelease = this.logicalKafkaVersionEntities.get(0);
+
+        for (int i = 1; i < this.logicalKafkaVersionEntities.size(); i++) {
+            if (latestRelease.compareTo(this.logicalKafkaVersionEntities.get(i)) < 0) {
+                latestRelease = this.logicalKafkaVersionEntities.get(i);
+            }
+        }
+
+        LOGGER.info("Latest release of Kafka is:{}", latestRelease.toString());
+
+        return latestRelease;
+    }
+
+    /**
+     * Gets the previous minor release.
+     * If this instance were 2.8.1 then at a certain point in time this method
+     * might return a version reflecting 2.7.2.
+     * On a later date it might return 2.7.3.
+     * It is allowed to cross major version boundaries.
+     * E.g. the previous minor release to 3.0.0 might be 2.8.2
```

Comment:
This isn't really explaining to the caller when the result might change. I know it's difficult to explain. I think you should say that this is relative to `latestRelease()`. Illustrating with a table of latestRelease and previousMinor might help.